### PR TITLE
TASK: remigrate all Neos.Neos fusion files and implementations

### PR DESCRIPTION
**Pull request by mhsdesign**
We noticed that Neos 9.0 had one missing feature in the Fusion implementations. To make sure there are no other faulty fusion implementations i checked and remigrated some fusion implementations.

Especially the ConvertUris implementation seems to have been affected. It seems as that it was now optimised for performance with a loss in features.

see https://github.com/neos/neos-development-collection/pull/4744

> it seems $someone tried to optimize the convert uris stuff from 8.3 while migrating the code to 9.0
> it should behave more performant with the cost of being less accurate and not covering all edgecases.
> 
> Maybe we should revert those performance optimization and introduce all those additional preg_match and replaces again for more accuracy. People might already have trouble enough with 9.0. Silent regressions like these will make things even harder.

@kitsunet wrote
> We can probably work on this at another point again, given that this is cached I don't see much problem in having this regex there, also I am sure this won't be in the top 5 of performance problems of rendering 🙈
>
> tl;dr; fine by me to go back to this.

Thats why i did checkout the 8.3 version of the `ConvertUrisImplementation` and remigrated its code to 9.0 without any additional performance in mind.

How i did it?

i reset the interesting fusion files, and checked what could possibly be wrong:

```
git checkout 8.3 -- Neos.Neos/Classes/Fusion/*Implementation.php
git checkout 8.3 -- Neos.Neos/Resources/Private/Fusion/
git checkout 8.3 -- Neos.Neos/Classes/Fusion/ExceptionHandlers/
git checkout 8.3 -- Neos.Neos/Classes/Fusion/Helper/
```
**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by mhsdesign** [comment]
https://github.com/neos/neos-development-collection/issues/4552 and https://github.com/neos/neos-development-collection/issues/4524 are the upcoming tasks to make the NodeUriImplementation work 100%

**Review by dlubitz — APPROVED** [review]
Looks good.

I wouldn't mind to take over some of the small improvements code wise (type hints, etc). But we can do that afterwards.
